client/lib/media/library-selected-store.js

```diff
@@ -89,9 +89,7 @@ MediaLibrarySelectedStore.getAll = function( siteId ) {
 		MediaLibrarySelectedStore
 			._media[ siteId ]
 			.map( itemId => MediaStore.get( siteId, itemId ) )
-			.filter( ( item ) => {
-				return ( item && item.guid );
-			} )
+			.filter( ( item ) => ( item && item.guid ) )
```

Comment:
As noted before, let's use something other than guid to verify that the item hasn't been deleted.